ghost/email-service/test/email-renderer.test.js

```diff
@@ -1182,6 +1186,68 @@ describe('Email renderer', function () {
                 clickTrackingEnabled: true
             };
 
+            renderedPost = '<p>Lexical Test</p><p><a href="https://external-domain.com/?ref=123">Hello</a><a href="https://encoded-link.com?code&#x3D;test">Hello</a><a href="https://example.com/?ref=123"><img src="example" /></a></p>';
```

Comment:
@rishabhgrg at which point to we get encoded urls into the system? Is it the user input or do we do it in the process of email delivery somewhere?

Reply:
@naz It happens somewhere in the mobiledoc renderer where we do mobiledoc -> html, so the url is already encoded by the time we parse the html for updating those links with our internal redirects. Since the source of truth for the link tracking is the url in HTML, this test starts from there with an encoded link and makes sure its redirected to the right thing.